sbt_common/storage/src/main/scala/uk/ac/wellcome/storage/type_classes/StorageStrategy.scala

```diff
@@ -0,0 +1,56 @@
+package uk.ac.wellcome.storage.type_classes
+
+import java.io.{ByteArrayInputStream, InputStream}
+
+import io.circe.Json
+
+import scala.io.Source.fromInputStream
+import scala.util.hashing.MurmurHash3
+
+trait StorageStrategy[T] {
+  def get(t: T): (InputStream, String)
+}
+
+object StorageStrategyGenerator {
+
+  private def hash(s: String) =
+    MurmurHash3
+      .stringHash(s, MurmurHash3.stringSeed)
+      .toHexString
+
+  def getKey[T](t: T)(implicit strategy: StorageStrategy[T]) =
+    strategy.get(t)
+
+  implicit val inputStreamKeyGetter: StorageStrategy[InputStream] =
```

Comment:
This is super-dumb I know, but now they are in the same place we can fix it later.